tcp_check/tests/test_tcp_check.py

```diff
@@ -109,6 +110,70 @@ def test_response_time(aggregator):
     assert len(aggregator.service_checks('tcp.can_connect')) == 1
 
 
+@pytest.mark.parametrize(
+    'hostname, getaddrinfo, expected_addrs',
+    [
+        pytest.param(
+            'localhost',
+            [
+                (socket.AF_INET6, socket.SOCK_STREAM, 6, '', ('::1', 80, 0, 0)),
+                (socket.AF_INET, socket.SOCK_STREAM, 6, '', ('127.0.0.1', 80)),
+            ],
+            '127.0.0.1',
+            id='localhost',
+        ),
+        pytest.param(
+            'some-hostname',
+            [
+                (socket.AF_INET, socket.SOCK_STREAM, 6, '', ('ip1', 80)),
+                (socket.AF_INET6, socket.SOCK_STREAM, 6, '', ('ip2', 80, 0, 0)),
+                (socket.AF_INET6, socket.SOCK_STREAM, 6, '', ('ip3', 80, 0, 0)),
+            ],
+            'ip1',
+            id='string hostname: ipv4 address 1st in list',
+        ),
+        pytest.param(
+            'another-hostname',
+            [
+                (socket.AF_INET6, socket.SOCK_STREAM, 6, '', ('ip1', 80, 0, 0)),
+                (socket.AF_INET, socket.SOCK_STREAM, 6, '', ('ip2', 80)),
+                (socket.AF_INET6, socket.SOCK_STREAM, 6, '', ('ip3', 80, 0, 0)),
+            ],
+            'ip2',
+            id='string hostname: ipv4 address 2nd in list',
+        ),
+    ],
+)
+def test_hostname_resolution(aggregator, hostname, getaddrinfo, expected_addrs):
+    """
+    Test that string hostnames get resolved to ipv4 address format properly.
+    """
+    instance = deepcopy(common.INSTANCE)
+    instance['host'] = hostname
+    instance['multiple_ips'] = False
+    check = TCPCheck(common.CHECK_NAME, {}, [instance])
+
+    with mock.patch('socket.getaddrinfo', return_value=getaddrinfo,), mock.patch(
+        'socket.gethostbyname', return_value=expected_addrs
+    ), mock.patch.object(check, 'connect', wraps=check.connect) as connect:
```

Comment:
Consider using the [`monkeypatch`](https://docs.pytest.org/en/6.2.x/monkeypatch.html) to remove this `with` which can be a bit hard to read.